lib/index.js

```diff
@@ -44,7 +44,10 @@ module.exports = function (source) {
 
   const isServer = target === 'node'
   const isShadow = !!options.shadowMode
-  const isProduction = options.productionMode || minimize || process.env.NODE_ENV === 'production'
+  const isProduction = options.productionMode || 
+                       minimize || 
+                       process.env.NODE_ENV === 'production' ||
+                       loaderContext._compilation.options.mode === 'production'
```

Comment:
is it really safe to access it like this?

Reply:
Hmm, looks like `_compilation` is deprecated: https://webpack.js.org/api/loaders/#this-_compilation
There's also an issue open on the Webpack repo: https://github.com/webpack/webpack/issues/6496
Maybe it should be done using the [plugin API](https://webpack.js.org/api/compiler-hooks/), if it's possible (not sure).